portal-web/test/functional/com/liferay/portalweb/macros/SAML.macro

```diff
@@ -215,21 +217,23 @@ definition {
 
 		// Configure IdP (Identity Provider) on specific Portal URL or default to current URL
 
-		if (isSet(idpURL)) {
-			if (isSet(userEmailAddress)) {
-				User.firstLoginUI(
-					password = "test",
-					specificURL = "${idpURL}",
-					userEmailAddress = "${userEmailAddress}");
-			}
-			else {
-				User.firstLoginUI(
-					password = "test",
-					specificURL = "${idpURL}",
-					userEmailAddress = "test@liferay.com");
-			}
+		Navigator.openSpecificURL(url = "${idpURL}");
```

Comment:
This macro requires the idpURL from default , I just tweaked it to check if we are already logged in or not to be able to skip the login process.
Are you suggesting to create another if else part with having a default url pointing to localhost? Because then we should extend the other macros here too, and I suggest we should do that in a different ticket

Reply:
Actually this macro doesn't require the idpURL as default. Generally we don't need to declare the idpURL if it's localhost:8080.
e.g [SAML#SPInitiatedSSOWithScreenNameIdentifier](https://github.com/liferay/liferay-portal/blob/master/portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/SAML.testcase#L2791)

In this situation, testcase is failed when you move Navigator.openSpecificURL(url = "${idpURL}"); in front of the if-else condition.

See details: https://testray.liferay.com/home/-/testray/case_results/1687966281

Reply:
Oh I see.. Let me check it out, as it should have failed even before I changed it if we don't put **idpURL** in it

Reply:
I pushed up a possilbe idea for the refactor of this